File: RoboLabWP/RoboLabWPComp/Model/Game/MZCellMatrix.cpp
#include "pch.h"
#include "MZCellMatrix.h"


MZCellMatrix::MZCellMatrix(int width, int height)
{
	_cells.resize(width);
    for(int i = 0; i < width; i++)
    {
		_cells[i].resize(height);
        for(int j = 0; j < height; j++)
        {
			_cells[i][j] = new MZCell();
        }
    }
}

MZCellMatrix::MZCellMatrix(MZCellMatrix &obj)
{
	_cells.resize(obj._cells.size());
	for(unsigned int i = 0; i < obj._cells.size(); i++)
    {
		_cells[i].resize(obj._cells[i].size());
        for(unsigned int j = 0; j < obj._cells[i].size(); j++)
        {
			_cells[i][j] = new MZCell(obj._cells[i][j]->wallsValue());
        }
    }
}

MZCellMatrix::~MZCellMatrix(void)
{
	int width = _cells.size();
	int height = _cells[0].size();
	for(int i = 0; i < width; i++)
    {
        for(int j = 0; j < height; j++)
        {
			delete(_cells[i][j]);
        }
    }
}

MZCell* MZCellMatrix::cellAtPosition(MZPosition * position)
{
	int cellsize = (int)_cells.size();
	int cellisize = (int)_cells[position->x()].size();
	if((position->x() >= 0) && (position->y() >=0) && (position->x() < cellsize) && (position->y() < cellisize))
    {
        return _cells[position->x()][position->y()];
    }
    return new MZCell();
}

File: RoboLabWP/RoboLabWP/Model/Game/MZCellMatrix.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RoboLabWP.Model.Basic_Types;
//
//  MZCellMatrix.cs
//  Maze
//
//  Класс, являющийся представлением двумерного массива объектов класса MZCell.

namespace RoboLabWP.Model.Game
{
    public class MZCellMatrix
    {
        private List<List<MZCell>> _cells;
         
        public MZCellMatrix(int width, int height)
        {
            _cells = new List<List<MZCell>>();
            for (int i = 0; i < width; i++)
            {
                _cells.Add(new List<MZCell>());
                for (int j = 0; j < height; j++)
                    _cells[i].Add(new MZCell());
            }
        }

        // вызврат клетки через позицию
        public MZCell cellAtPosition(MZPosition position)
        {
            if (position.x() >= 0 && position.y() >= 0 && position.x() < _cells.Count() && position.y() < _cells[0].Count())
            {
                return _cells[(int)position.x()][(int)position.y()];
            }
            return new MZCell();
        }
    }
}
